src/System.Security.Principal.Windows/src/System/Security/Principal/SID.cs

```diff
@@ -792,21 +792,36 @@ public override string ToString()
         {
             if (_sddlForm == null)
             {
-                StringBuilder result = new StringBuilder();
-
                 //
                 // Typecasting of _IdentifierAuthority to a long below is important, since
                 // otherwise you would see this: "S-1-NTAuthority-32-544"
                 //
-
-                result.Append("S-1-").Append((long)_identifierAuthority);
-
-                for (int i = 0; i < SubAuthorityCount; i++)
+                // length of buffer calculation
+                // prefix = "S-1-".Length: 4;
+                // authority: long.MaxValue.ToString("D") : 19;
+                // subauth = MaxSubAuthorities * ( uint.MaxValue.ToString("D").Length + '-'.Length ): 15 * (10+1): 165;
+                // max possible length = 4 + 19 + 165: 188
+                Span<char> result = stackalloc char[188];
+                int written;
+                int[] values = _subAuthorities;
+                result[0] = 'S';
+                result[1] = '-';
+                result[0] = '1';
+                result[1] = '-';
+                int length = 4;
+                ((long)_identifierAuthority).TryFormat(result.Slice(4), out written);
+                length += written;
+
+                for (int index = 0; index < _subAuthorities.Length; index++)
```

Comment:
Use `values.Legth` instead? It has the cached value of `_subAuthorities` ?

Reply:
Done.